backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/impexp/BankStatementImportProcess.java

```diff
@@ -175,62 +180,147 @@ private void createBankStatementLine(final I_I_BankStatement importRecord)
 	}
 
 	private BankStatementId createBankStatement(@NonNull final I_I_BankStatement importBankStatement)
 	{
 		return bankStatementDAO.createBankStatement(BankStatementCreateRequest.builder()
-				.orgId(OrgId.ofRepoId(importBankStatement.getAD_Org_ID()))
-				.orgBankAccountId(BankAccountId.ofRepoId(importBankStatement.getC_BP_BankAccount_ID()))
-				// TODO add documentNo to import
-				.statementDate(TimeUtil.asLocalDate(importBankStatement.getStatementDate()))
-				.name(importBankStatement.getName())
-				.description(importBankStatement.getDescription())
-				.eft(BankStatementCreateRequest.ElectronicFundsTransfer.builder()
-						.statementDate(TimeUtil.asLocalDate(importBankStatement.getEftStatementDate()))
-						.statementReference(importBankStatement.getEftStatementReference())
-						.build())
-				.build());
+															.orgId(OrgId.ofRepoId(importBankStatement.getAD_Org_ID()))
+															.orgBankAccountId(BankAccountId.ofRepoId(importBankStatement.getC_BP_BankAccount_ID()))
+															// TODO add documentNo to import
+															.statementDate(TimeUtil.asLocalDate(importBankStatement.getStatementDate()))
+															.name(importBankStatement.getName())
+															.description(importBankStatement.getDescription())
+															.eft(BankStatementCreateRequest.ElectronicFundsTransfer.builder()
+																		 .statementDate(TimeUtil.asLocalDate(importBankStatement.getEftStatementDate()))
+																		 .statementReference(importBankStatement.getEftStatementReference())
+																		 .build())
+															.build());
+	}
+
+	@VisibleForTesting
+	static void computeUpdateLineAmts(@NonNull final I_I_BankStatement importRecord)
+	{
+		final String amtFormat = getAmtFormat(importRecord);
+
+		if (X_I_BankStatement.AMTFORMAT_AmountPlusIndicator.equals(amtFormat))
+		{
+			final BigDecimal trxAmt;
+			final boolean isDebit = "S".equals(importRecord.getDebitOrCreditIndicator());
+			if (isDebit)
+			{
+				trxAmt = importRecord.getDebitOrCreditAmt().negate();
+			}
+			else
+			{
+				trxAmt = importRecord.getDebitOrCreditAmt();
+			}
+
+			importRecord.setTrxAmt(trxAmt);
+			final BigDecimal stmtAmt = trxAmt.add(importRecord.getChargeAmt()).add(importRecord.getInterestAmt());
+			importRecord.setStmtAmt(stmtAmt);
+		}
+		else if (X_I_BankStatement.AMTFORMAT_DebitPlusCredit.equals(amtFormat))
+		{
+			final BigDecimal stmtAmt = importRecord.getCreditStmtAmt().subtract(importRecord.getDebitStmtAmt());
+			importRecord.setStmtAmt(stmtAmt);
+			final BigDecimal trxAmt = stmtAmt.subtract(importRecord.getInterestAmt()).subtract(importRecord.getChargeAmt());
+			importRecord.setTrxAmt(trxAmt);
+		}
+		else if (X_I_BankStatement.AMTFORMAT_Straight.equals(amtFormat))
+		{
+			importRecord.setStmtAmt(importRecord.getTrxAmt());
+		}
+
+		{
+			final BigDecimal sum = importRecord.getTrxAmt().add(importRecord.getInterestAmt()).add(importRecord.getChargeAmt());
```

Comment:
what case is this?
Shouldn't we throw exception for this case?

Reply:
This is the case where our amt calculations are wrong, and we are throwing an error: 
if `StmtAmt != TrxAmt + ChargeAmt + InterestAmt` then throw exception.

Reply:
ahh, got it. thx for explaining.
 it's fine!